common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java

```diff
@@ -54,7 +54,7 @@ public int getPort() {
 
   @Override
   public int hashCode() {
-    return host.hashCode();
+    return id.hashCode();
```

Comment:
Why not include all three fields (`id`, `host`, `port`) in hashCode()?
Such as `Objects.hash(id, host, port)`.

Reply:
By default `id = host + "-" + port`, so id is enough.

Reply:
> By default `id = host + "-" + port`, so id is enough.

Should we remove the constructor at line 37?

Reply:
No. In some cases, the id does not need to be recalculated.

Reply:
OK. Can you add some comments here to make future reviewer be aware of it.

Reply:
Done